libs/domains/clusters/feature/src/lib/cluster-type/cluster-type.tsx

```diff
@@ -0,0 +1,26 @@
+import { type CloudProviderEnum, KubernetesEnum } from 'qovery-typescript-axios'
+import { P, match } from 'ts-pattern'
+import { Badge, type BadgeProps } from '@qovery/shared/ui'
+
+export interface ClusterTypeProps extends Omit<BadgeProps, 'color'> {
+  cloudProvider: keyof typeof CloudProviderEnum
+  kubernetes?: KubernetesEnum
+}
+
+export function ClusterType({ cloudProvider, kubernetes, ...props }: ClusterTypeProps) {
+  const clusterType = match([cloudProvider, kubernetes])
+    .with(['AWS', KubernetesEnum.K3_S], () => 'EC2 (K3S)')
+    .with(['AWS', KubernetesEnum.MANAGED], ['AWS', undefined], () => 'Managed (EKS)')
```

Comment:
Because `kubernetes` is optional param.
In fact, the combination `cloudProvider=AWS` + `kubernetes=undefined` is not possible functionally.
So this is just to comply with ts-pattern `exhaustive` typechecking